paddle/platform/gpu_info.cc

```diff
@@ -75,13 +74,6 @@ size_t GpuMaxChunkSize() {
 
   GpuMemoryUsage(available, total);
 
-  if (IsEnvVarDefined(kEnvFractionGpuMemoryToUse)) {
```

Comment:
I don't think that we need `platform/environment.{h,cc}`, which looks too heavy.  This segment of code could be 

```c++
if (std::getenv(kEnvFractionGpuMemoryToUse)) {
  *FLAG_fraction_of_gpu_memory_to_use = std::strtod(std::getenv(kEnvFractionGpuMemoryToUse), nullptr);
  PADDLE_ENFORCE_GT(val, 0.0);		
  PADDLE_ENFORCE_LE(val, 1.0);
}
```